client/app/scripts/superdesk-search/search.js

```diff
@@ -1048,6 +1073,55 @@
             };
         }])
 
+        .directive('sdSaveSearch', ['$location', 'asset', 'api', 'session', 'notify', 'gettext', 
+            function($location, asset, api, session, notify, gettext) {
+            return {
+                templateUrl: asset.templateUrl('superdesk-search/views/save-search.html'),
+                link: function(scope, elem) {
+                    scope.edit = null;
+
+                    scope.editItem = function() {
+                        scope.edit = _.create(scope.editingSearch) || {};
+                    };
+
+                    scope.saveas = function() {
+                        scope.edit = _.clone(scope.editingSearch) || {};
+                        delete scope.edit._id;
+                    };
+
+                    scope.cancel = function() {
+                        scope.edit = null;
+                    };
+
+                    scope.clear = function() {
+                        scope.cancel();
+                        $location.url('/search');
+                    };
+
+                    scope.save = function(editSearch) {
+
+                        function onSuccess() {
+                            notify.success(gettext('Saved search is saved successfully'));
+                            scope.cancel();
+                            scope.changeTab();
+                        }
+
+                        function onFail() {
+                            notify.error(gettext('Error. Saved search could not be saved.'));
+                        }
+
+                        editSearch.filter = {query: $location.search()};
+                        if (!editSearch._id) {
+                            api('saved_searches', session.identity).save({}, editSearch).then(onSuccess, onFail);
+                        } else {
+                            api('saved_searches', session.identity).save(scope.editingSearch, editSearch).then(onSuccess, onFail);
```

Comment:
IMO code duplication can be reduced here, the only difference is the value of the first argument. It probably makes sense to determine its value first based on the `editSearch._id` and then have a single `api.save(...)` line.